src/tests/routes/controllers/auth.test.js

```diff
@@ -0,0 +1,97 @@
+import chai from 'chai';
+import chaiHttp from 'chai-http';
+
+import db from '../../../db/models';
+import app from '../../../server';
+
+chai.use(chaiHttp);
+const { expect } = chai;
+const { User } = db;
+
+
+const userCredentials = {
+  email: 'giant@gmail.com',
+  password: 'smiley007',
+  username: 'gentlejane',
+};
+
+const nonExistent = {
+  email: 'humpty@dumpty.com',
+  password: 'not12database',
+};
+
+const { email, password } = userCredentials;
+
+const url = '/api/v1/auth';
+const loginUrl = '/api/v1/auth/login';
+
+
+describe('User registration', () => {
+  before(async () => {
+    await User.sync({ force: true });
+  });
+
+  context('CREATE USER', () => {
+    it('should create a user successfully when valid input are supplied', async () => {
+      const res = await chai.request(app)
+        .post(url)
+        .send(userCredentials);
+      expect(res.status).to.equal(201);
+      expect(res.body).to.have.property('message').to.be.a('String');
+    });
+  });
+});
+
+
+describe('User Login Feature', () => {
+  before(async () => {
+    try {
+      await User.sync({ force: true });
+      return User.create(userCredentials);
+    } catch (error) {
+      return error.toString();
+    }
+  });
+
+  context('User can log in', () => {
+    it('should return a token', async () => {
+      const res = await chai.request(app)
+        .post(loginUrl)
+        .send({
+          email,
+          password,
+        });
+      expect(res.status).to.equal(200);
+      expect(res.body).to.have.property('data').to.be.a('String');
```

Comment:
can we test for what is stored in the data variable instead of just testing the type of data.